components/favicon_title_handler/favicon_title_handler.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {injectIntl, IntlShape} from 'react-intl';
+
+import {Channel} from 'mattermost-redux/types/channels';
+import {Team} from 'mattermost-redux/types/teams';
+
+import * as UserAgent from 'utils/user_agent';
+import {Constants} from 'utils/constants';
+
+import favicon16x16 from 'images/favicon/favicon-16x16.png';
+import favicon32x32 from 'images/favicon/favicon-32x32.png';
+import favicon96x96 from 'images/favicon/favicon-96x96.png';
+import redDotFavicon16x16 from 'images/favicon/favicon-reddot-16x16.png';
+import redDotFavicon32x32 from 'images/favicon/favicon-reddot-32x32.png';
+import redDotFavicon96x96 from 'images/favicon/favicon-reddot-96x96.png';
+
+type Props = {
+    intl: IntlShape;
+    unreads: {
+        messageCount: number;
+        mentionCount: number;
+    };
+    siteName?: string;
+    currentChannel: Channel;
+    currentTeam: Team;
+    currentTeammate: Channel | null;
+};
+
+type State = {
+
+};
+
+class FaviconTitleHandler extends React.PureComponent<Props, State> {
+    badgesActive: boolean;
+    lastBadgesActive: boolean;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.badgesActive = false;
+        this.lastBadgesActive = false;
+    }
+
+    componentDidUpdate() {
+        this.updateTitle();
+        this.setBadgesActiveAndFavicon();
+    }
+
+    setBadgesActiveAndFavicon() {
+        if (!(UserAgent.isFirefox() || UserAgent.isChrome())) {
+            return;
+        }
+
+        const link = document.querySelector('link[rel="icon"]');
+
+        if (!link) {
+            return;
+        }
+
+        this.lastBadgesActive = this.badgesActive;
```

Comment:
Do we need to store these values on the component, or could we just derive `lastBadgesActive` from the previous props?

Reply:
We can probably derive from previous props, i'll make the change.